packages/hdwallet-xdefi/src/xdefi.ts

```diff
@@ -0,0 +1,285 @@
+  }
+
+  public describePath(msg: core.DescribePath): core.PathDescription {
+    return this.info.describePath(msg);
+  }
+
+  public async getPublicKeys(msg: Array<core.GetPublicKey>): Promise<Array<core.PublicKey | null>> {
+    // Ethereum public keys are not exposed by the RPC API
+    return [];
+  }
+
+  public async isInitialized(): Promise<boolean> {
+    return true;
+  }
+
+  public async disconnect(): Promise<void> {}
+
+  public async ethSupportsNetwork(chainId: number = 1): Promise<boolean> {
+    return chainId === 1;
+  }
+
+  public async ethSupportsSecureTransfer(): Promise<boolean> {
+    return false;
+  }
+
+  public ethSupportsNativeShapeShift(): boolean {
+    return false;
+  }
+
+  public async ethSupportsEIP1559(): Promise<boolean> {
+    return true;
+  }
+
+  public ethGetAccountPaths(msg: core.ETHGetAccountPath): Array<core.ETHAccountPath> {
+    return eth.ethGetAccountPaths(msg);
+  }
+
+  public ethNextAccountPath(msg: core.ETHAccountPath): core.ETHAccountPath | undefined {
+    return this.info.ethNextAccountPath(msg);
+  }
+
+  public async ethGetAddress(): Promise<string | null> {
+    if (this.ethAddress) {
+      return this.ethAddress;
+    }
+    const address = await eth.ethGetAddress(this.provider);
+    if (address) {
+      this.ethAddress = address;
+      return address;
+    } else {
+      this.ethAddress = null;
+      return null;
+    }
+  }
+
+  public async ethSignTx(msg: core.ETHSignTx): Promise<core.ETHSignedTx | null> {
+    const address = await this.ethGetAddress();
+    return address ? eth.ethSignTx(msg, this.provider, address) : null;
+  }
+
+  public async ethSendTx(msg: core.ETHSignTx): Promise<core.ETHTxHash | null> {
+    const address = await this.ethGetAddress();
+    return address ? eth.ethSendTx(msg, this.provider, address) : null;
+  }
+
+  public async ethSignMessage(msg: core.ETHSignMessage): Promise<core.ETHSignedMessage | null> {
+    const address = await this.ethGetAddress();
+    return address ? eth.ethSignMessage(msg, this.provider, address) : null;
+  }
+
+  public async ethVerifyMessage(msg: core.ETHVerifyMessage): Promise<boolean | null> {
+    return eth.ethVerifyMessage(msg, this.provider);
+  }
+
+  public async getDeviceID(): Promise<string> {
+    return "xDeFi:" + (await this.ethGetAddress());
+  }
+
+  public async getFirmwareVersion(): Promise<string> {
+    return "xDeFi";
+  }
+}
+
+export class XDeFiHDWalletInfo implements core.HDWalletInfo, core.ETHWalletInfo {
+  readonly _supportsETHInfo = true;
+
+  public getVendor(): string {
+    return "XDeFi";
+  }
+
+  public hasOnDevicePinEntry(): boolean {
+    return false;
+  }
+
+  public hasOnDevicePassphrase(): boolean {
+    return false;
+  }
+
+  public hasOnDeviceDisplay(): boolean {
+    return false;
+  }
+
+  public hasOnDeviceRecovery(): boolean {
+    return false;
+  }
+
+  public hasNativeShapeShift(): boolean {
+    // It doesn't... yet?
```

Comment:
(It never will, this "native shapeshift" thing has to do with dead legacy stuff from when ShapeShift was an exchange)
```suggestion
```